### PR TITLE
Throw custom error on too many login attempts

### PR DESCRIPTION
**Pull request by sagarmulchandani**
## What does this PR do ?
Fix / Change for #1781 
Added a new Error for exceeded login attempts after checking rate limiting.

_Note: This PR is part of my contribution towards `Hacktoberfest`. If upon review it is legible please do add `hacktoberfest-accepted` label to it. Thank you :)_

### CODE REVIEW

**Comment by Aschen** [comment]
Hi @sagarmulchandani and thanks you very much for your contribution!

We are happy to accept your PR for the Hacktoberfest ;-) 

Before merging we need the tests to pass, please consider the review I've just made

File: lib/api/funnel.js [hunk 1]
```diff
@@ -269,6 +269,10 @@ class Funnel {
             })
             .then(allowed => {
               if (!allowed) {
+                const { controller, action } = _request.input;
+                if (controller === 'auth' && action === 'login') {
```
Comment (Aschen) [inline lib/api/funnel.js:273]: Missing unit tests for this case

File: lib/kerror/codes/2-api.json [hunk 2]
```diff
@@ -128,6 +128,12 @@
           "code": 9,
           "message": "Admin user is already set.",
           "class": "PreconditionError"
+        },
+        "too_many_login_requests": {
+          "description": "The request has been refused because logins attempts per second has been exceeded for this user",
+          "code": 10,
+          "message": "Rejected: requests for logins per second exceeded for this user.",
+          "class": "TooManyLoginRequestsError"
```
Comment (Aschen) [inline lib/kerror/codes/2-api.json:136]: This property is used to construct the error and must be the name of one of our error classes.  
Those errors are defined in [another repository](https://github.com/kuzzleio/kuzzle-common-objects/tree/master/lib/errors).  
You can use the `TooManyRequestsError` class for example:
```suggestion
          "class": "TooManyRequestsError"
```

**Review by Aschen — CHANGES_REQUESTED** [review]

**Comment by sagarmulchandani** [comment]
Hi @Aschen , Thank you for getting back on this so quickly. Will be making and changes and updating soon.

File: lib/api/funnel.js (hunk 1 above)
Comment (Aschen) [reply lib/api/funnel.js:273]: @sagarmulchandani you can duplicate this unit test and change the `request.input.controller` and `request.input.action` property to simulate a `auth:login` request:
https://github.com/kuzzleio/kuzzle/blob/master/test/api/funnel/execute.test.js#L132

Comment (sagarmulchandani) [reply lib/api/funnel.js:273]: Hi @Aschen , have made the changes but facing an issue with running `npm test` and have been occupied a little with my office work to look into this. Will be fixing this and updating soon. Apologies for the delay on my end.

Comment (Aschen) [reply lib/api/funnel.js:273]: You can join us on [Discord](http://join.discord.kuzzle.io) if you need some help with the tests

**Pull request closed without merging**